pyrsa/util/vis_utils.py

```diff
@@ -0,0 +1,450 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Collection of helper methods for vis module
+
++ Weighted_MDS:       an MDS class that incorporates weighting
++ weight_to_matrices: batch squareform() to weight matrices
+
+@author: baihan
+
+Notice: 
+
+The functions of MDS in this module are modified from 
+the Python package scikit-learn, originally written by
+Nelle Varoquaux <nelle.varoquaux@gmail.com> under BSD 
+licence <https://en.wikipedia.org/wiki/BSD_licenses>. 
+We modified the MDS function to include an additional
+functionality of having an important matrix as an input.
+"""
+
+import numpy as np
+import warnings
+from sklearn.base import BaseEstimator
+from sklearn.metrics import euclidean_distances
+from sklearn.utils import check_random_state, check_array, check_symmetric
+from sklearn.externals.joblib import Parallel
+from sklearn.externals.joblib import delayed
```

Comment:
We don't currently depend on Joblib yet, so we'd have to add this to the requirements file (currently this would fail with an import error I think)

Reply:
I thought if we already have sklearn as a requirement, wouldn't joblib be implicitly already also installed during pyrsa and sklearn installations?

Reply:
You are right, I just tried and joblib installs along with sklearn. Hm I don't like joblib so much because it doesn't work well with out HPC cluster. It does some sneaky stuff that no-one fully understands. 
But I guess we have to accept it.

Reply:
I just updated the code to include joblib as a requirement. In addition, I removed two sklearn dependencies to lighten things up.

Reply:
Wait, so which one would you prefer? I just saw your comment popping up after my latest commit (there is some delay in Github). I guess we can also potentially go back to this issue later, to try to move away from joblib if people find it troublesome from time to time~